### ISSUE TITLE
credhub metrics are cached indefinitly

### ISSUE THREAD

**Issue by poblin-orange** [orange-cloudfoundry/credhub_exporter#11]
using prometheus bosh release 25 / credhub exporter 0.1.6, we have applied a soon to expire cert rotation, which should have discarded the alert.

The credhub_exporter keeps giving to prometheus the old metrics (metrics seemed cached with no time limit).

```
credhub_certificate_expires_at{deployment="fe-prod",environment="fe-prod",id="08a91662-0ad0-40a5-a4c4-86a01e83cdd4",index="1",name="director_ssl-ca",path="/bosh-master/bosh-coab/director_ssl"} 1.563376859e+09
17/7/2019 à 15:20:59 
credhub_certificate_expires_at{deployment="fe-prod",environment="fe-prod",id="11117042-0ddf-47e9-ad41-416dc3e5522b",index="1",name="director_ssl-ca",path="/bosh-master/bosh-coab/director_ssl"} 1.877630181e+09
1/7/2029 à 19:56:21  
```

Restarting the credhub_exporter jobs solved the issue.
cc @ogrand

**Comment by gberche-orange** [comment]
Fixed in version [0.1.7](https://github.com/orange-cloudfoundry/credhub_exporter/releases/tag/v0.1.7) through  #13, thanks a lot @Freakin !

@psycofdj is mentionning privately to me that he is currently unable to prioritize further testing this PR and contributing the bump upstream similar to https://github.com/bosh-prometheus/prometheus-boshrelease/pull/251

@Freaking please feel free to go ahead and submit such a bump PRupstream if this can help speed up the resolution and its deployment in your prometheus deployments.

/CC @poblin-orange @ogrand

**Comment by psycofdj** [comment]
[v0.2.0](https://github.com/orange-cloudfoundry/credhub_exporter/releases/tag/v0.2.0) now resets the metrics on each analyze loop
